packages/ifpack2/src/Ifpack2_Details_Chebyshev_def.hpp

```diff
@@ -638,16 +639,35 @@ setParameters (Teuchos::ParameterList& plist)
 
   // We don't want to fill these parameters in, because they shouldn't
   // be visible to Ifpack2::Chebyshev users.
+  if (plist.isParameter ("chebyshev: kind")) {
+    chebyshevKind = plist.get<std::string> ("chebyshev: kind");
+    TEUCHOS_TEST_FOR_EXCEPTION(
+      chebyshevKind != "first" &&
+      chebyshevKind != "textbook" &&
+      chebyshevKind != "fourth" &&
+      chebyshevKind != "opt_fourth",
+      std::invalid_argument,
+      "Ifpack2::Chebyshev: Ifpack2 only supports \"first\", \"textbook\", \"fourth\", and \"opt_fourth\", for \"chebyshev: kind\".");
+  }
+
+#ifndef IFPACK2_ENABLE_DEPRECATED_CODE
   if (plist.isParameter ("chebyshev: textbook algorithm")) {
-    textbookAlgorithm = plist.get<bool> ("chebyshev: textbook algorithm");
+    const bool textbookAlgorithm = plist.get<bool> ("chebyshev: textbook algorithm");
```

Comment:
What about the case of "chebyshev: textbook algorithm" being in the parameter list and its value being `false`? I guess that should results in `chebyshevKind = "first"`?

Reply:
Not necessarily -- what if a user sets both:

```
"chebyshev: kind", "fourth"
"chebyshev: textbook algorithm", "false" 
```

I would think that this would mean `chebyshevKind = "fourth"`.

At the same time, I should also explicitly guard against:
```
"chebyshev: kind", "fourth" // or any other setting that is not textbook
"chebyshev: textbook algorithm", "true" 
```

since it's not clear to a user which should take precedence.

```
"chebyshev: kind", "textbook"
"chebyshev: textbook algorithm", "true" 
```

should also be okay -- the user just specified more-or-less the same option twice.

However,

```
"chebyshev: kind", "textbook"
"chebyshev: textbook algorithm", "false" 
```

should throw an error -- how are we supposed to interpret that?

:face_exhaling: All of this additional logic for a boolean knob that will (hopefully) soon be deprecated.

Reply:
:-) 

I think what we really want is that someone with an old input deck does not see a change of behavior. They would not have "chebyshev: kind" in their input deck, and it seems that no default value would get injected prior to those lines of code. They might however have "chebyshev: textbook algorithm".

Once the user specifies "chebyshev: kind" I feel it's safe to ignore "chebyshev: textbook algorithm".